Modules/_ctypes/cfield.c

```diff
@@ -1057,10 +1056,10 @@ f_set_sw(void *ptr, PyObject *value, Py_ssize_t size)
     if (x == -1 && PyErr_Occurred())
         return NULL;
 #ifdef WORDS_BIGENDIAN
```

Comment:
Can the `le` argument of `PyFloat_{Un}Pack*` be replaced with three-value `endianness` ( `big`, `little`, and `native`)? The `native` option would allow to move all `#ifdef WORDS_BIGENDIAN` fiddling inside the (up)pack functions.

Reply:
If this parameter is changed, it will be way harder to provide this function in pythoncapi_compat to older Python versions: https://github.com/pythoncapi/pythoncapi_compat/pull/26 And it maybe become more complicated to write a single code base supporting old and new Python versions.

It's easy to implement the "native" behavior by checking if your system uses little endian or not with your preferred way.

* marshal uses little endian
* struct supports big and little endian
* pickle uses big endian
* ctypes supports big and little endian
* msgpack uses big endian

So far, I didn't see a module using the "native endian". For a serialization function, it's not convenient.

Reply:
I wrote https://github.com/python/cpython/pull/31832 to suggest using PY_BIG_ENDIAN macro to use the native endian.